### PR TITLE
Backend refactoring

### PR DESCRIPTION
**Pull request by karenl1**
Get rid of for-looped queries in API by replacing them with more specific SQL queries in model

### CODE REVIEW

File: public/js/bookshelf.js [hunk 1]
```diff
@@ -360,7 +366,7 @@ $(document).ready(function(){
 			console.log("remove book id", book_to_remove_id);
 
 			// remove wanted books from wish list
```
Comment (jerrylinew) [inline public/js/bookshelf.js]: Let's make this consistent and change all instances of wish list to what it should be

File: public/js/bookshelf.js [hunk 2]
```diff
@@ -360,7 +366,7 @@ $(document).ready(function(){
...
-			$.post("/api/wish_list/remove", { user_id: "user", book_id: book_to_remove_id },
+			$.post("api/possible_trades/remove", { user_id: "user", owned_book_id: owned_book_id, wanted_book_id: book_to_remove_id },
 				function(data){
 					if(data.status === 0){
 						console.log('successfully removed wanted book from wish_list');
```
Comment (jerrylinew) [inline public/js/bookshelf.js]: same here

**Comment by jayendrajog** [comment]
In the future, can you split this up into multiple PRs? Some of these functions are definitely not dependent on each other and could have been reviewed separately, and it'll make code review + catching mistakes much easier in general

**Comment by karenl1** [comment]
@jayendrajog you right my b next time will split into separate PRs 👍

**Comment by jayendrajog** [comment]
@codeKaren lgtm 

also sry i guess i never created the PR to remove wish_list, thanks for taking care of that

**Pull request merged**